src/mpi/coll/src/csel.c

```diff
@@ -577,17 +577,31 @@ int MPIR_Csel_create_from_buf(const char *json,
 int MPIR_Csel_create_from_file(const char *json_file,
                                void *(*create_container) (struct json_object *), void **csel_)
 {
+    int mpi_errno = MPI_SUCCESS;
+
     MPIR_Assert(strcmp(json_file, ""));
 
     int fd = open(json_file, O_RDONLY);
+    if (fd == -1) {
+        fprintf
+            (stderr,
+             "ERROR: Opening MPIR_CVAR_COLL_SELECTION_TUNING_JSON_FILE failed.\n"
+             "ERROR: Please check the file at %s\n"
+             "ERROR: or unset MPIR_CVAR_COLL_SELECTION_TUNING_JSON_FILE to use the default settings.\n",
+             json_file);
+        MPIR_ERR_SETANDJUMP3(mpi_errno, MPI_ERR_INTERN, "**open", "**open %s %d %d", json_file,
+                             MPIR_Process.rank, fd);
```

Comment:
How about creating an `errorname` and rolling the message into it?

Reply:
Sure thing. Here is the new output:
```
Abort(471455248): Fatal error in internal_Init: Internal MPI error!, error stack:
internal_Init(45897)...........: MPI_Init(argc=0x16d9bb054, argv=0x16d9bb048) failed
MPII_Init_thread(182)..........:
MPII_Coll_init(143)............:
MPIR_Csel_create_from_file(586):  Unable to open collective tuning file /tmp/foobar.
Check the file or unset MPIR_CVAR_COLL_SELECTION_TUNING_JSON_FILE to use the default settings.
```